rawnode.go

```diff
@@ -428,6 +438,10 @@ func (rn *RawNode) acceptReady(rd Ready) {
 		}
 	}
 	rn.raft.msgs = nil
+	// Release the msgsAfterAppend slice and put it back into the msgsAfterAppendPool
+	if cap(rn.raft.msgsAfterAppend) > 0 {
+		rn.raft.msgsAfterAppendPool.Put(rn.raft.msgsAfterAppend[:0])
```

Comment:
nit: add a comment explaining the logic of this optimization. Similar to what you've said (https://github.com/etcd-io/raft/pull/51#issuecomment-1546221117)